app/src/main/java/fr/free/nrw/commons/contributions/ContributionsActivity.java

```diff
@@ -342,4 +354,57 @@ public static void startYourself(Context context) {
         Intent contributionsIntent = new Intent(context, ContributionsActivity.class);
         context.startActivity(contributionsIntent);
     }
+
+    private void displayFeedbackPopup() {
+
+        Date strDate = null;
+        try {
+            String validUntil = "23/08/2017";
+            SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+            strDate = sdf.parse(validUntil);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        final SharedPreferences prefs =  PreferenceManager.getDefaultSharedPreferences(
+                CommonsApplication.getInstance());
+
+        // boolean to save users request about displaying popup
+        boolean displayFeedbackPopup = prefs.getBoolean("display_feedbak_popup", true);
+
+        // boolean to recognize is application re-started. Will be used for "remind me later" option
+        int appStartCounter = prefs.getInt("app_start_counter" ,0);
+
+        // if time is valid and shared pref says display
+        if (new Date().before(strDate) && displayFeedbackPopup && (appStartCounter == 4)) {
+
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setTitle(getResources().getString(R.string.feedback_popup_title));
+            builder.setMessage(getResources().getString(R.string.feedback_popup_description));
+            builder.setPositiveButton(getResources().getString(R.string.feedback_popup_accept),
+                    new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int which) {
+                            // Go to the page
+                            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri
+                                    .parse(getResources()
+                                    .getString(R.string.feedback_page_url)));
+                            startActivity(browserIntent);
+                            // No need to dislay this window to the user again.
+                            prefs.edit().putBoolean("display_feedbak_popup" , false).commit();
+                            dialog.dismiss();
+                        }
+                    });
+            builder.setNegativeButton(getResources().getString(R.string.feedback_popup_decline),
+                    new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            // Dismiss the dialog and not to show it later
+                            prefs.edit().putBoolean("display_feedbak_popup", false).commit();
+                            dialog.dismiss();
+                        }
+                    });
+            AlertDialog alert = builder.create();
+            alert.show();
```

Comment:
This is totally a _personal_ preference, but since this is a `Builder` I prefer to chain the calls together and reduce the repetition of "builder." at the start of all the lines.  😄  Taken to it's  fullest, you dont even need the local variables as everything chains all the way down through the `create()` and the `show()`.